File: src/main/scala/com/gu/support/workers/states/CheckoutFailureState.scala
package com.gu.support.workers.states

import com.gu.support.workers.User
import com.gu.support.workers.CheckoutFailureReasons.CheckoutFailureReason

case class CheckoutFailureState(user: User, checkoutFailureReason: CheckoutFailureReason) extends StepFunctionUserState

File: support-workers/src/typescript/model/checkoutFailureState.ts
import { z } from 'zod';
import { checkoutFailureReasons } from '../errors/checkoutFailureReasons';
import { userSchema } from './stateSchemas';

export const checkoutFailureReasonSchema = z.enum(checkoutFailureReasons);

export const checkoutFailureStateSchema = z.object({
	user: userSchema,
	checkoutFailureReason: checkoutFailureReasonSchema,
});

export type CheckoutFailureState = z.infer<typeof checkoutFailureStateSchema>;
